app/src/main/java/com/google/jetpackcamera/ui/permissions/PermissionsViewModel.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.ui.permissions
+
+import androidx.compose.runtime.mutableStateListOf
+import androidx.lifecycle.ViewModel
+
+/**
+ * A [ViewModel] for [PermissionsScreen]]
+ */
+class PermissionsViewModel(
+    val permissionEnums: Set<PermissionEnum>
+) : ViewModel() {
+
+    val visiblePermissionDialogQueue = mutableStateListOf<PermissionEnum>()
```

Comment:
For consistency, let's keep `compose` classes out of the `ViewModel`. Instead, you can replace this with a `MutableStateFlow<PermissionsUiState>` where `PermissionsUiState` looks something like:
```
sealed interface PermissionsUiState {
  data class PermissionsNeeded(val currentPermission: PermissionEnum) : PermissionsUiState 
  object AllPermissionsGranted : PermissionsUiState
}
```

You will need to move some of the bookkeeping from JcaApp to PermissionsViewModel, but that's probably the right thing to do anyways.